lib/arclight/traject/ead2_config.rb

```diff
@@ -303,14 +303,16 @@
     accumulator << context.output_hash['parent_ssm'].last
   end
 
-  to_field 'parent_unittitles_ssm' do |_record, accumulator, context|
-    ## Top level document
+  to_field 'parent_unittitles_ssm' do |_rec, accumulator, context|
+    # top level document
     accumulator.concat context.clipboard[:parent].output_hash['normalized_title_ssm']
-    ## Other components
-    context.output_hash['parent_ssm']&.drop(1)&.each do |id|
-      accumulator.concat Array
-        .wrap(context.clipboard[:parent].output_hash['components'])
-        .select { |c| c['ref_ssi'] == [id] }.map { |c| c['normalized_title_ssm'] }.flatten
+    parent_ssm = context.output_hash['parent_ssm']
+    components = context.clipboard[:parent].output_hash['components']
+
+    # other components
+    if parent_ssm && components
+      ancestors = parent_ssm.drop(1).map { |x| [x] }
+      accumulator.concat components.select { |c| ancestors.include? c['ref_ssi'] }.flat_map { |c| c['normalized_title_ssm'] }
```

Comment:
👍 thank you for this, these lines seem to be identical to me.

I am curious as to why the speed was improved - perhaps because `#flatten` was originall invoked after the results of `.select { |c| c['ref_ssi'] == [id] }.map` were issued?  I can't imagine that introducing `parent_ssm.drop(1).map { |x| [x] }` could have changed much.